lib/string_basics.dart

```diff
@@ -200,6 +201,46 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a truncated version of the string.
+  ///
+  /// Example:
+  /// ```dart
+  /// final sentence = 'The quick brown fox jumps over the lazy dog';
+  /// final truncated = sentence.truncate(20); // 'The quick brown fox...'
+  /// ```
+  ///
+  /// The [length] is the truncated length of the string.
+  /// The [substitution] is the substituting string of the truncated characters.
+  /// If not null or empty it will be appended at the end of the truncated string.
+  /// The [trimTrailingWhitespace] is whether or not to trim the spaces of the truncated string
+  /// before appending the ending string.
+  /// The [includeSubstitutionLength] is whether or not that the length of the substitution string will be included
+  /// with the intended truncated length.
+  String truncate(
+    int length, {
+    String substitution = '',
+    bool trimTrailingWhitespace = true,
+    bool includeSubstitutionLength = false,
```

Comment:
Yeah, I agree this is better. But since it's already so long anyway, how about `includeSubstitutionInLength`? I feel like that makes it even more explicit.

Reply:
Updated the parameter name.

Reply:
Thanks!